v2/fb-xcom-api-validator/src/auth-provider.ts

```diff
@@ -0,0 +1,74 @@
+import { randomUUID } from "crypto"
+import config from "./config";
+import { getVerifyKey, signerFactory } from "./signing";
+import { encoderFactory } from "./encoding";
+
+
+export type HTTPMethod = "GET" | "POST" | "PUT" | "DELETE";
+
+
+const authConfig = config.get('authentication');
+
+
+export function getSecurityHeaders(method: HTTPMethod, endpoint: string, body: any, timestamp?: number, nonce?: string): Record<string, string> {
+    if (!nonce) {
+        nonce = randomUUID();
+    }
+    if (!timestamp) {
+        timestamp = Date.now();
+    }
+    const signature = getRequestSignature(method, endpoint, body, timestamp, nonce);
+
+    return {
+        "X-FBAPI-KEY": authConfig.apiKey,
+        "X-FBAPI-SIGNATURE": encodeURIComponent(signature),
```

Comment:
I'd prefer all the config values to be supplied here to `getRequestSignature` and not be buried deep inside the call stack — it makes testing more difficult.